instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/PluginImplUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class PluginImplUtil {
+  private PluginImplUtil() {}
+
+  private static final Logger logger = Logger.getLogger(PluginImplUtil.class.getName());
+
+  /**
+   * Check if the given {@code moduleNameImpl} is present.
+   *
+   * <p>For library instrumentations, the Impls will always be available but might fail to load if
+   * the corresponding SDK classes are not on the class path. For javaagent, the Impl is available
+   * only when the corresponding InstrumentationModule was successfully applied (muzzle passed).
+   *
+   * <p>Note that an present-but-incompatible library can only be detected by Muzzle. In
+   * library-mode, users need to ensure they are using a compatible SDK (component) versions
+   * themselves.
+   *
+   * @param implSimpleClassName The simple name of the impl class, e.g. {@code "SqsImpl"}. *
+   */
+  static boolean isImplPresent(String implSimpleClassName) {
+    String implFullClassName =
+        PluginImplUtil.class.getName().replace(".PluginImplUtil", "." + implSimpleClassName);
```

Comment:
This is intentionally using the more awkward way, since getPackage might return `null` in some cases (we don't know which class loader the application we are running in is using). See https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8603#discussion_r1230528720

Should I add a comment?

Reply:
Yes please!

Reply:
e53bf957ef5eeb89da840cd9ebf194ea4d5dc85b